Redesign domain search input field

DL6ER opened the pull request:
# What does this implement/fix?

### Before

![Screenshot from 2024-02-12 12-35-50](https://github.com/pi-hole/web/assets/16748619/5c422782-52e2-4705-8fe8-1247d13e537f)


### Now

![Screenshot from 2024-02-12 12-38-21](https://github.com/pi-hole/web/assets/16748619/546c85ca-a390-4d20-a0a5-9b37197dda6a)


**Related issue or feature (if applicable):** Fixes #2701 

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

DL6ER committed:
Redesign search input field to solve #2701

Signed-off-by: DL6ER <dl6er@dl6er.de>

rdwebdesign commented:
Now that you changed the design to use a single button with shorter text, you can remove these CSS rules:

https://github.com/pi-hole/web/blob/938c6ef383a19ec65557399d80b311b8ee02ce9a/style/pi-hole.css#L565-L577

**EDIT:**
After out conversation, I added a commit to remove the unused CSS.

rdwebdesign committed:
Remove unused CSS rules

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Better text and controls layout

- use 4-8 columns instead of 6-6;
- use label before the number control (this will show the label on the previous line, when there is a line break);
- right-align second column in larger screens and add small margin;
- remove old HTML comments.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

PromoFaux reviewed (approved):
Tested OK
